.github/workflows/ci.yml

```
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main   
+  pull_request:  
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+          components: clippy
```

The action has been archived a year ago or so, let's use `dtolnay/rust-toolchain`.

```suggestion
      - name: Install Rust
        uses: dtolnay/rust-toolchain@master
        with:
          toolchain: stable
          components: clippy
```